### ISSUE TITLE
[Bug] [Dashboard] Dashboard.log spammy error message

### ISSUE THREAD

**Issue by simon-mo** [ray-project/ray#19200]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Dashboard

### What happened + What you expected to happen

```

2021-10-07 11:00:00,324 ERROR base_events.py:1619 -- Exception in callback PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()
handle: <Handle PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()>
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/completion_queue.pyx.pxi", line 147, in grpc._cython.cygrpc.PollerCompletionQueue._handle_events
BlockingIOError: [Errno 11] Resource temporarily unavailable
2021-10-07 11:00:00,327 ERROR base_events.py:1619 -- Exception in callback PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()
handle: <Handle PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()>
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/completion_queue.pyx.pxi", line 147, in grpc._cython.cygrpc.PollerCompletionQueue._handle_events
BlockingIOError: [Errno 11] Resource temporarily unavailable
2021-10-07 11:00:00,330 ERROR base_events.py:1619 -- Exception in callback PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()
handle: <Handle PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()>
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/completion_queue.pyx.pxi", line 147, in grpc._cython.cygrpc.PollerCompletionQueue._handle_events
BlockingIOError: [Errno 11] Resource temporarily unavailable
2021-10-07 11:00:00,334 ERROR base_events.py:1619 -- Exception in callback PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()
handle: <Handle PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()>
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/completion_queue.pyx.pxi", line 147, in grpc._cython.cygrpc.PollerCompletionQueue._handle_events
BlockingIOError: [Errno 11] Resource temporarily unavailable
2021-10-07 11:00:00,336 ERROR base_events.py:1619 -- Exception in callback PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()
handle: <Handle PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()>
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/completion_queue.pyx.pxi", line 147, in grpc._cython.cygrpc.PollerCompletionQueue._handle_events
BlockingIOError: [Errno 11] Resource temporarily unavailable
2021-10-07 11:00:00,653 ERROR base_events.py:1619 -- Exception in callback PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()
handle: <Handle PollerCompletionQueue._handle_events(<_UnixSelecto...e debug=False>)()>
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/completion_queue.pyx.pxi", line 147, in grpc._cython.cygrpc.PollerCompletionQueue._handle_events
BlockingIOError: [Errno 11] Resource temporarily unavailable
```
it is logged per second. I can't find out were does it come from so would need to bisect the commits. 

### Reproduction script

```
import ray

ray.init()

@ray.remote
def f():
    import time
    time.sleep(100000)

ray.get(f.remote())
```

then tail dashboard.log from /tmp/raysession_latest/logs

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by simon-mo** [comment]
```
git bisect start
# bad: [80990d18c7a4d33ab982dafe626178779b2b478c] [C++ API] support head_args config in C++ API (#18709)
git bisect bad 80990d18c7a4d33ab982dafe626178779b2b478c
# good: [77fb273c61dbab2ab2b93101e8649f4eabe7d6ad] [Doc] Update macos nightly wheel names (#17813)
git bisect good 77fb273c61dbab2ab2b93101e8649f4eabe7d6ad
# good: [2b520bafc5508a6e7f4786d9017412826acab9d8] [release/alert] less results (#17637)
git bisect good 2b520bafc5508a6e7f4786d9017412826acab9d8
# good: [d3786ac131989bf3a66b4515cddf37d413b4633a] Bump Java version to 2.0.0-SNAPSHOT (#15394)
git bisect good d3786ac131989bf3a66b4515cddf37d413b4633a
# bad: [f2d8f23fb640218adb336d3b7da5e4d08c225cac] [workflow] Define default `__getstate__` and `__setstate__` (#18459)
git bisect bad f2d8f23fb640218adb336d3b7da5e4d08c225cac
# bad: [c4199a80544d67971a15c9f2a8b6b48f76086bfd] Add more workflow comparisons (#18347)
git bisect bad c4199a80544d67971a15c9f2a8b6b48f76086bfd
# good: [673bf35c1f2d1af6ff5b92bf857ffd980bafeb87] Refactor BackendState to be per-backend instead of global (#18255)
git bisect good 673bf35c1f2d1af6ff5b92bf857ffd980bafeb87
# good: [814095add682df98fc83dd1f1d5bb16ce94b7613] Revert "Change instance type for some tests (#18248)" (#18320)
git bisect good 814095add682df98fc83dd1f1d5bb16ce94b7613
# good: [ac5d255c9cf6402187a00c2cfbe546977bf003e1] [rllib/docker] silent unzip of atari roms (#18340)
git bisect good ac5d255c9cf6402187a00c2cfbe546977bf003e1
# good: [e049d52d299cf68bf378c35d1a402299ba71ea7c] Retry application-level error by default for datasets (#18296)
git bisect good e049d52d299cf68bf378c35d1a402299ba71ea7c
# bad: [cf4fb4edb3816888663af3450b3533cd46785cb5] [Core][plasma] fix the data race issue (#18312)
git bisect bad cf4fb4edb3816888663af3450b3533cd46785cb5
# bad: [e61160d514b72295a74ffc884f4b0e77520d836c] [Dashboard] Move gcs health check to a separate thread to avoid crashing due to excessive CPU usage. (#18236)
git bisect bad e61160d514b72295a74ffc884f4b0e77520d836c
# first bad commit: [e61160d514b72295a74ffc884f4b0e77520d836c] [Dashboard] Move gcs health check to a separate thread to avoid crashing due to excessive CPU usage. (#18236)
```

of course it's my own commits!
```
e61160d514b72295a74ffc884f4b0e77520d836c is the first bad commit
commit e61160d514b72295a74ffc884f4b0e77520d836c
Author: Simon Mo <simon.mo@hey.com>
Date:   Fri Sep 3 14:23:56 2021 -0700

    [Dashboard] Move gcs health check to a separate thread to avoid crashing due to excessive CPU usage. (#18236)

 dashboard/consts.py                             |   1 +
 dashboard/head.py                               | 113 +++++++++++++++---------
 dashboard/modules/log/tests/test_log.py         |   4 +-
 dashboard/tests/conftest.py                     |   2 +
 python/ray/scripts/scripts.py                   |   2 +-
 src/ray/gcs/gcs_server/gcs_heartbeat_manager.cc |   1 -
 src/ray/protobuf/gcs_service.proto              |   2 -
```

**Comment by matthewdeng** [comment]
Hey @simon-mo what's the status of this? Will this be going into 1.7.1? Should we hold off on 1.8 branch cut until this is resolved?
